### PR TITLE
Add linear probing analysis

### PR DESCRIPTION
**Pull request by pfgryz**

### CODE REVIEW

File: scripts/train_linear_probing.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    # arr = np.array(batch["activation"], dtype=float)
+    # norms = np.linalg.norm(arr, axis=1, keepdims=True)
+    # batch["activation"] = (arr / norms).tolist()
+
+    return batch
+
+
+def encode_sae(model: AutoEncoderTopK, batch: dict, device: str) -> dict:
+    tensor_in = torch.tensor(batch["activation"]).to(device)
+    batch["activation"] = model.encode(tensor_in).detach().cpu().numpy()
```
Comment (MikolajSzawerda) [inline scripts/train_linear_probing.py:80]: ae.encode(tensor_in, use_threshold=True)

File: scripts/train_linear_probing.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+def add_label(batch: dict, label: int) -> dict:
+    batch["label"] = [label] * len(batch["activation"])
+    return batch
+
+
+def unpack_and_normalize(batch: dict) -> dict:
+    batch["activation"] = [item[0] for item in batch["activation"]]
+
+    # arr = np.array(batch["activation"], dtype=float)
+    # norms = np.linalg.norm(arr, axis=1, keepdims=True)
```
Comment (MikolajSzawerda) [inline scripts/train_linear_probing.py:72]: Hm, for me such normalization don't make sense, it should be scaled by global value, not local one to batch

File: scripts/train_linear_probing.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            / str(cfg.layer_id)
+            / cfg.dataset.split
+            / "*.arrow"
+        ),
+        streaming=False,
+        num_proc=8,
+        split="train",
+    )
+
+    ds_negative = ds_negative.map(lambda x: add_label(x, 0), num_proc=12, batched=True, batch_size=1024)
```
Comment (MikolajSzawerda) [inline scripts/train_linear_probing.py:146]: ds_negative.add_column("label", [0] * ds_negative.num_rows)
